prov/efa/src/rdm/efa_rdm_pke.h

```diff
@@ -222,15 +191,12 @@ struct efa_rdm_pke {
 	 *       (thus data has been copied to wiredata).
 	 *    b) packet is an incoming (RX) packet.
 	 */
-	char wiredata[0];
+	_Alignas(EFA_RDM_PKE_ALIGNMENT) char wiredata[0];
 };
 
+
 #if defined(static_assert) && defined(__x86_64__)
-#if ENABLE_DEBUG
-static_assert(sizeof(struct efa_rdm_pke) == 256, "efa_rdm_pke check");
-#else
-static_assert(sizeof(struct efa_rdm_pke) == 128, "efa_rdm_pke check");
-#endif
+static_assert(sizeof (struct efa_rdm_pke) % EFA_RDM_PKE_ALIGNMENT == 0, "efa_rdm_pke alignment check");
```

Comment:
Do we need to make EFA_RDM_PKE_ALIGNMENT 256 when ENABLE_DEBUG=1?

Reply:
No. This change makes the size of `struct efa_rdm_pke` 128 bytes regardless of whether debug mode is enabled or not. The extra `struct dlist_entry` when debug mode is enabled is currently smaller than the unused space added by `_Alignas(128)`.

In the event the preceding struct members exceed 128 bytes, `wiredata` would be at the 256 byte offset, since `_Alignas(128)` specifies it be aligned to the nearest 128 byte boundary ([Try it online!](https://tio.run/##jY9dC4IwFIav5684FIEDEwoKywj6HRUy5lYD28LNG6PfbmciiH1fDZ7nnPc949MT500zVpoXVS5gY12uTHzeBgPECnXSLzQX8oUp7TzDBy5M6ZDCLSDWlRV3II3JZvPEE1LhRJI5yNroi9Buj@qYosp2HjEbIqDAz6yEnDm29/KeBoO4@WL5OW71TxwhAbmWuC7DkVW1MBKG565hUlcRGCmtcGg7HPuQ1h30KIJuNRzu0n7vyUTtEZSm39rxd@/aEf9ux6EP7Wj69nvTPAA)). So `EFA_RDM_PKE_ALIGNMENT` should remain as the intended alignment for `wiredata`

Reply:
Then why was there a `uint8_t pad[112]` in the `ENABLE_DEBUG` condition? It seems like we were trying to align to 256 when ENABLE_DEBUG=1

Reply:
But performance doesn't matter in debug mode, so I think it's OK to align to 128 all the time

Reply:
>Then why was there a uint8_t pad[112] in the ENABLE_DEBUG condition? It seems like we were trying to align to 256 when ENABLE_DEBUG=1

It was aimed to make both debug and non-debug builds to be 128 aligned. There are more members introduced in debug build which makes efa_rdm_pke exceed 128 bytes, so we seek for the next 128 aligned integer (256)

Reply:
We are safe as long as the total size of efa_rdm_pke can be multiples of 128 bytes for both builds